base/pmap.jl

```diff
@@ -0,0 +1,102 @@
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+
+"""
+    pgenerate([::WorkerPool], f, c...) -> iterator
+
+Apply `f` to each element of `c` in parallel using available workers and tasks.
+
+For multiple collection arguments, apply f elementwise.
+
+Results are returned in order as they become available.
+
+Note that `f` must be made available to all worker processes; see
+[Code Availability and Loading Packages](:ref:`Code Availability
+and Loading Packages <man-parallel-computing-code-availability>`)
+for details.
+"""
+function pgenerate(p::WorkerPool, f, c)
+    if length(p) == 0
+        return asyncgenerate(f, c)
+    end
+    batches = batchsplit(c, min_batch_count = length(p) * 3)
+    return flatten(asyncgenerate(remote(p, b -> asyncmap(f, b)), batches))
+end
+
+pgenerate(p::WorkerPool, f, c1, c...) = pgenerate(p, a->f(a...), zip(c1, c...))
+
+pgenerate(f, c) = pgenerate(default_worker_pool(), f, c...)
+pgenerate(f, c1, c...) = pgenerate(a->f(a...), zip(c1, c...))
+
+
+
+"""
+    pmap([::WorkerPool], f, c...) -> collection
+
+Transform collection `c` by applying `f` to each element using available
+workers and tasks.
+
+For multiple collection arguments, apply f elementwise.
+
+Note that `f` must be made available to all worker processes; see
+[Code Availability and Loading Packages](:ref:`Code Availability
+and Loading Packages <man-parallel-computing-code-availability>`)
+for details.
+"""
+pmap(p::WorkerPool, f, c...) = collect(pgenerate(p, f, c...))
+#pmap(f, c...) = pmap(default_worker_pool(), f, c...)
+
+function pmap(f, c...; err_retry=nothing, err_stop=nothing, pids=nothing)
```

Comment:
Question: is there any precedence for having a depwarn message that can be suppressed by e.g. a global variable.

e.g.
```julia
    if !haskey(ENV, "JULIA_SUPPRESS_PMAP_WARNING")
        depwarn("`err_retry` and `err_stop` no longer have any effect" *
                "use `pmap(retry(f), c)` or `pmap(@catch(f, c))`." *
                "The default retry and error handling behaviour has changed. See ? pmap." *
                "set ENV["JULIA_SUPPRESS_PMAP_WARNING"] to suppress this warning".)
    end
```